compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -758,20 +758,26 @@ impl<'a> Parser<'a> {
 
         // If the user typed "..==" instead of "..=", we want to give them
         // a specific error message telling them to use "..=".
+        // If they typed "..=>", suggest they use ".. =>".
         // Otherwise, we assume that they meant to type a half open exclusive
         // range and give them an error telling them to do that instead.
-        if matches!(tok.kind, token::Eq) && tok.span.lo() == span.hi() {
-            let span_with_eq = span.to(tok.span);
+        let no_space = tok.span.lo() == span.hi();
```

Comment:
is `span` always coming from `self.prev_token.span`? If so, could we drop the `span` argument to this fn?